### PR TITLE
Add `From` implementation to `ExtrinsicSigner`

### LINKED ISSUE #452: impl From<sp_core::sr25519::Pair> for ExtrinsicSigner

### PR DESCRIPTION
**Pull request by masapr**
closes #452

### CODE REVIEW

File: primitives/src/signer.rs [hunk 1]
```diff
@@ -90,3 +90,37 @@ where
...
+	#[test]
+	fn test_extrinsic_signer_from_sr25519_pair() {
+		let alice: sr25519::Pair = Pair::from_string(
+			"0xe5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0a",
+			None,
+		)
+		.unwrap();
+
+		let es_from =
+			ExtrinsicSigner::<sr25519::Pair, sr25519::Signature, Runtime>::from(alice.clone());
```
Comment (haerdib) [inline primitives/src/signer.rs:120]: I don't think that's what was imagined in #452. The cool thing about `From` is that a simple `.into()` does everything for you, no need to call any complex function.

An example can be found here, where the public address is simply converted to a Runtime::Account:
https://github.com/scs/substrate-api-client/blob/027ed593d0907255c2c2460ed9b49ad3f7205143/primitives/src/signer.rs#L61

However, I do not know if that's even possible.

Comment (masapr) [reply primitives/src/signer.rs]: Thanks, makes sense! (... still learning, so wasn't aware of that ...)

**Review by haerdib — APPROVED** [review]

File: primitives/Cargo.toml [hunk 2]
```diff
@@ -29,6 +29,7 @@ pallet-staking = { optional = true, git = "https://github.com/paritytech/substra
 
 [dev-dependencies]
 node-template-runtime = { git = "https://github.com/paritytech/substrate.git", branch = "master" }
+kitchensink-runtime = { git = "https://github.com/paritytech/substrate.git", branch = "master" }
```
[the remark this answers is not in the document]
Comment (masapr) [reply primitives/Cargo.toml]: less dependencies are always better -> I changed it

File: primitives/src/signer.rs [hunk 3]
```diff
@@ -90,3 +90,37 @@ where
...
+
+	#[test]
+	fn test_extrinsic_signer_from_sr25519_pair() {
+		let alice: sr25519::Pair = Pair::from_string(
+			"0xe5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0a",
+			None,
+		)
+		.unwrap();
+
+		let es_converted: ExtrinsicSigner<sr25519::Pair, sr25519::Signature, Runtime> =
```
[the remark this answers is not in the document]
Comment (masapr) [reply primitives/src/signer.rs]: Thanks, implemented it

**Pull request merged**